src/Entity/DeliveryRepository.php

```diff
@@ -87,4 +106,23 @@ public function findOneByHashId(string $hashId)
 
         return $this->find($id);
     }
+
+    public function searchWithSonic(QueryBuilder $qb, string $q, string $locale, ?Store $store = null)
+    {
+        $search = new \Psonic\Search($this->sonicClient);
+        $search->connect($this->sonicSecretPassword);
+
+        $collection = (null !== $store) ? sprintf('store:%d:deliveries', $store->getId()) : 'store:*:deliveries';
+
+        $ids = $search->query($collection, $this->sonicNamespace,
+            $q, $limit = null, $offset = null, Languages::getAlpha3Code($locale));
```

Comment:
When there are lots of results, this may break due to lack of memory. Pagination should be implemented.